### ISSUE TITLE
Complete Backend Code

### ISSUE THREAD

**Issue by jcjordyn130** [jcjordyn130/IntroToSEProjectFall2024#13]
- [ ] Implement all SRS Requirements
- [ ] Implement AT LEAST 3 out of 4 requirements from the Product Topic Document

**Comment by jcjordyn130** [comment]
I completed the database design mostly to the previous sprint's diagrams. I changed it a little bit as the diagrams were impossible to implement as written in Python with SQLite3. I also completed the entire Database class, including the Database types.

I'm currently working on the backend API and have basic user login/logout working along with proper error management and API key management (in house, the existing implementations are too complex).

**Comment by jcjordyn130** [comment]
I committed the back end code to the repository with commit 90b5371288d07304242112789fb8f11f27723e3b.

After the previous comment, I completed user management entirely and got about 75% done with implementing order management.

All that's left to do is to implement product management and payment methods, then cleanup the code a little.
 
User privilege separation (Buyer, seller, and admin) all work as expected, and is kept track of via the API key given after login.
